src/metabase/models/persisted_info.clj

```diff
@@ -78,32 +79,59 @@
              (assoc card :persisted (contains? existing-ids id)))
            cards))))
 
-(defn mark-for-deletion!
-  "Marks PersistedInfo as `deletable`, these will at some point be cleaned up by the PersistPrune task."
-  [conditions-map]
-  (db/update-where! PersistedInfo conditions-map :active false, :state "deletable", :state_change_at :%now))
+(defn mark-for-pruning!
+  "Marks PersistedInfo as `deletable` or `off`, these will at some point be cleaned up by the PersistPrune task.
 
-(defn make-ready!
+   `deletable` will wipe out all trace of persisted-info and allow them to be turned back on by automatic processes
+     use when you are disabling peristence at a high level.
+   `off` will ensure automatic processes do not pick up these up and re-enable."
+  ([conditions-map]
+   (mark-for-pruning! conditions-map "deletable"))
+  ([conditions-map state]
+   (db/update-where! PersistedInfo conditions-map :active false, :state state, :state_change_at :%now)))
+
+(defn- create!
   "Marks PersistedInfo as `creating`, these will at some point be persisted by the PersistRefresh task."
   [user-id card]
   (let [slug (-> card :name slug-name)
         {:keys [database_id]} card
-        card-id (u/the-id card)
+        card-id (u/the-id card)]
+    (db/insert! PersistedInfo {:card_id         card-id
+                               :database_id     database_id
+                               :question_slug   slug
+                               :table_name      (format "model_%s_%s" card-id slug)
+                               :active          false
+                               :refresh_begin   :%now
+                               :refresh_end     nil
+                               :state           "creating"
+                               :state_change_at :%now
+                               :creator_id      user-id})))
+
+(defn ready-unpersisted-models!
+  "Looks for all models in database and creates a persisted-info ready to be synced."
+  [database-id]
+  (let [cards (db/select Card {:where [:and
+                                       [:= :database_id database-id]
+                                       [:= :dataset true]
+                                       [:not [:exists {:select [1]
+                                                       :from [:persisted_info]
+                                                       :where [:= :persisted_info.card_id :report_card.id]}]]]})]
+    (doseq [card cards]
+      (create! nil card))))
```

Comment:
should this be a bulk insert rather than repeated single inserts?

Reply:
agree, fixed